packages/discount-code-importer/src/main.js

```diff
@@ -1,3 +1,221 @@
+      // No catch block as errors will be caught in the CLI
+  }
+
+  _processBatches (codes: CodeDataArray) {
+    // Batch to `batchSize` to reduce necessary fetch API calls
+    const batchedList = _.chunk(codes, this.batchSize)
+    return Promise.mapSeries(batchedList, (codeObjects: CodeDataArray) => {
+      // Build predicate and fetch existing code
+      const predicate = DiscountCodeImport._buildPredicate(codeObjects)
+      const service = this._createService()
+      const uri = service
+        .where(predicate)
+        .perPage(this.batchSize)
+        .build()
+      return this.client.execute({
+        uri,
+        method: 'GET',
+      })
+        .then(({ body: { results: existingCodes } }: Object) => (
+          this._createOrUpdate(codeObjects, existingCodes)
+        ))
+    })
+      .then(() => Promise.resolve(this.summaryReport()))
+      .catch(error => Promise.reject({
+        error,
+        summary: this._summary,
+      }),
+      )
+  }
+
+  _createOrUpdate (newCodes: CodeDataArray, existingCodes: CodeDataArray) {
+    return Promise.map(newCodes, (newCode: CodeData) => {
+      const existingCode = _.find(existingCodes, ['code', newCode.code])
+      if (existingCode)
+        return this._update(newCode, existingCode)
+          .then((response) => {
+            if (response && response.statusCode === 304)
+              this._summary.unchanged += 1
+            else
+              this._summary.updated += 1
+            return Promise.resolve()
+          })
+          .catch((error) => {
+            if (this.continueOnProblems) {
+              this._summary.updateErrorCount += 1
+              this._summary.errors.push(error)
+              // eslint-disable-next-line max-len
+              const msg = 'Update error occured but ignored. See summary for details'
+              this.logger.warn(msg)
+              return Promise.resolve()
+            }
+            this._summary.updateErrorCount += 1
+            this._summary.errors.push(error)
+            return Promise.reject(error)
+          })
+      return this._create(newCode)
+        .then(() => {
+          this._summary.created += 1
+          return Promise.resolve()
+        })
+        .catch((error) => {
+          if (this.continueOnProblems) {
+            this._summary.createErrorCount += 1
+            this._summary.errors.push(error)
+            // eslint-disable-next-line max-len
+            const msg = 'Create error occured but ignored. See summary for details'
+            this.logger.warn(msg)
+            return Promise.resolve()
+          }
+          this._summary.createErrorCount += 1
+          this._summary.errors.push(error)
+          return Promise.reject(error)
+        })
+    })
+  }
+
+  _update (newCode: CodeData, existingCode: CodeData) {
+    const actions = this.syncDiscountCodes.buildActions(newCode, existingCode)
+    // don't call API if there's no update action
+    if (!actions.length)
```

Comment:
That's the current implementation. unchanged gets incremented if there's no update action